log/logtest/recorder_test.go

```diff
@@ -115,17 +115,29 @@ func TestRecorderEmitAndReset(t *testing.T) {
 
 	r1 := log.Record{}
 	r1.SetSeverity(log.SeverityInfo)
-	l.Emit(context.Background(), r1)
-	assert.Equal(t, r.Result()[0].Records, []log.Record{r1})
+	ctx := context.Background()
+
+	l.Emit(ctx, r1)
+	assert.Equal(t, r.Result()[0].Records, []EmittedRecord{
+		{r1, ctx},
+	})
 
 	nl := r.Logger("test")
 	assert.Empty(t, r.Result()[1].Records)
 
 	r2 := log.Record{}
 	r2.SetSeverity(log.SeverityError)
-	nl.Emit(context.Background(), r2)
-	assert.Equal(t, r.Result()[0].Records, []log.Record{r1})
-	assert.Equal(t, r.Result()[1].Records, []log.Record{r2})
+	// We want a non-background context here so it's different from `ctx`.
+	ctx2, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	nl.Emit(ctx2, r2)
+	assert.Equal(t, r.Result()[0].Records, []EmittedRecord{
+		{r1, ctx},
+	})
+	assert.Equal(t, r.Result()[1].Records, []EmittedRecord{
+		{r2, ctx2},
+	})
```

Comment:
nit: it would be safer (for backwards compatibility) to assert the records using helper functions as https://pkg.go.dev/go.opentelemetry.io/otel/sdk/log#Record is not really comparable. Here we rely on reflection magic of testify.

you could already reuse https://github.com/open-telemetry/opentelemetry-go/blob/724c24085ebf3c52c953f6b25b18d9c0ef2b3b86/log/logtest/factory_test.go#L70-L88

Reply:
I have added an `assertRecord`.
I wonder if we shouldn't make it public though, as it seems like something many bridges could benefit from.

Reply:
> I wonder if we shouldn't make it public though, as it seems like something many bridges could benefit from.

I thought about the same. If you already see places where it could be used then I am OK with exporting it as a separate PR.